cpp/src/io/parquet/reader_impl_preprocess_utils.cu

```diff
@@ -247,23 +248,23 @@ void generate_depth_remappings(
               kernel_error::to_string(error));
   }
 
-  for (auto& chunk : chunks) {
-    total_pages += chunk.num_data_pages + chunk.num_dict_pages;
+  for (auto chunk = chunks.host_begin(); chunk != chunks.host_end(); chunk++) {
```

Comment:
looks like `hostdevice_span` has a different API for host `begin`/`end` than `hostdevice_vector`. Should we make these consistent?